src/vsr/journal.zig

```diff
@@ -669,117 +885,379 @@ pub fn Journal(comptime Replica: type, comptime Storage: type) type {
+        ///    power failure.
+        /// 8! Journal recovery again — but now slot 8 is loaded *without* being marked faulty.
+        ///    So we may incorrectly nack slot 8's message.
+        ///
+        /// Therefore, recovery will never load a header into a slot *and* mark that slot faulty.
+        ///
+        ///
+        /// B) When replica_count=1, repairing broken/lost prepares over VSR is not an option,
+        /// so if a message is faulty the replica will abort.
+        ///
+        ///
+        /// Recovery decision table:
+        ///
+        ///   label                   @A  @B  @C  @D  @E  @F  @G  @H  @I  @J  @K  @L
+        ///   header valid             0   1   1   0   0   1   1   1   1   1   1   1
+        ///   header reserved          _   1   0   _   _   1   0   1   0   0   0   0
+        ///   prepare valid            0   0   0   1   1   1   1   1   1   1   1   1
+        ///   prepare reserved         _   _   _   1   0   0   1   1   0   0   0   0
+        ///   match checksum           _   _   _   _   _   _   _  !1   0   0   0   1
+        ///   match op                 _   _   _   _   _   _   _  !1   <   >   1  !1
+        ///   match view               _   _   _   _   _   _   _  !1   _   _  !0  !1
+        ///   decision (replicas>1)  vsr vsr vsr vsr vsr vsr vsr nil fix vsr vsr eql
+        ///   decision (replicas=1)                  fix fix
+        ///
+        /// Legend:
+        ///
+        ///    0  false
+        ///    1  true
+        ///   !0  assert false
+        ///   !1  assert true
+        ///    _  ignore
+        ///    <  header.op < prepare.op
+        ///    >  header.op > prepare.op
+        ///
+        /// A "valid" header/prepare:
+        /// 1. has a valid checksum
+        /// 2. has the correct cluster
+        /// 3. is in the correct slot (op % slot_count)
+        /// 4. has command=reserved or command=prepare
+        ///
+        ///
+        /// Regarding cases @E and @F for *single-replica clusters*:
+        ///
+        /// Most likely the redundant header write was corrupted/lost (respectively).
+        /// Since single-replica clusters never rewrite ops, it can be locally repaired by copying
+        /// over the prepare's header.
+        ///
+        /// However, another (astronomically low) possibility is this:
+        ///
+        /// 1. A prepare is written to slot 12.
+        /// 2. A redundant header is written to slot 12.
+        /// 3. Continue normal operation; wrap around the WAL.
+        /// 4. A prepare is written to slot 12, but lost.
+        /// 5. A redundant header is written to slot 12.
+        /// 6. Replica crash.
+        /// 7. On recovery, the old prepare (from step 1) is read and used to "repair" the corrupt
+        ///    redundant header.
+        ///
+        /// This would be a safety violation — we assume that the likelihood of this event is
+        /// negligible.
```

Comment:
I worked through this and I think this is actually neither `@E` nor `@F` but actually `@J` (`header.op > prepare.op`) where we would go to `.vsr` for a cluster-of-one and do the right thing and crash?